From: Xin Hao <xhao@linux.alibaba.com>
Subject: mm/damon/sysfs: more simplified code calls damon_target_has_pid()
Date: Fri, 16 Sep 2022 21:35:35 +0800

This patch further simplifies the damon_target_has_pid() call and makes
the damon_sysfs_destroy_targets() look cleaner.

Link: https://lkml.kernel.org/r/20220916133535.7428-1-xhao@linux.alibaba.com
Fixes: 49fb890735ab (mm/damon/sysfs: avoid call damon_target_has_pid() repeatedly)
Signed-off-by: Xin Hao <xhao@linux.alibaba.com>
Suggested-by: SeongJae Park <sj@kernel.org>
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/sysfs.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/mm/damon/sysfs.c~mm-damon-sysfs-more-simplified-code-calls-damon_target_has_pid
+++ a/mm/damon/sysfs.c
@@ -2143,10 +2143,7 @@ static int damon_sysfs_set_attrs(struct
 static void damon_sysfs_destroy_targets(struct damon_ctx *ctx)
 {
 	struct damon_target *t, *next;
-	bool has_pid = false;
-
-	if (damon_target_has_pid(ctx))
-		has_pid = true;
+	bool has_pid = damon_target_has_pid(ctx);
 
 	damon_for_each_target_safe(t, next, ctx) {
 		if (has_pid)
_
